### PR TITLE
Create new unit tests for version and kind validation

### PR DESCRIPTION
**Pull request by spzala**
This is a follow up PR per discussion in the
https://github.com/kubernetes/kubernetes/pull/53587
Creating new unit tests here for basic and aggregated validation of version
and kind group.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/util/openapi/validation/validation_test.go [hunk 1]
```diff
@@ -335,4 +335,63 @@ items:
...
+		err := validator.ValidateBytes([]byte(`
+kind: Pod
+metadata:
+  name: name
+spec:
+  containers:
+  - name: name
+    image: image
+`))
+		Expect(err.Error()).To(Equal("apiVersion not set"))
```
Comment (apelisse) [inline pkg/kubectl/cmd/util/openapi/validation/validation_test.go]: I'm not super fan of testing errors by string, can you look at how it's done for other errors?

File: pkg/kubectl/cmd/util/openapi/validation/validation_test.go [hunk 2]
```diff
@@ -335,4 +335,63 @@ items:
...
+    name: name
+  spec:
+    type: NodePort
+    ports:
+    - port: 123
+      targetPort: 1234
+      name: name
+    selector:
+      name: name
+`))
```
Comment (apelisse) [inline pkg/kubectl/cmd/util/openapi/validation/validation_test.go]: you don't have a test for the feature you just implemented, do you?

Comment (spzala) [reply pkg/kubectl/cmd/util/openapi/validation/validation_test.go]: @apelisse hi, actually yes I do. The last two tests here both shows failure with aggregated errors are towards the features we recently merged. i.e. "([apiVersion isn't string type, kind not set]"  and "[kind not set, apiVersion not set]") ] .. LOC 352 and 396. Hope makes sense. Thanks!

Comment (apelisse) [reply pkg/kubectl/cmd/util/openapi/validation/validation_test.go]: But it happens in two different objects. I think that test would have worked without your change? Maybe I'm wrong though.

Comment (apelisse) [reply pkg/kubectl/cmd/util/openapi/validation/validation_test.go]: OK, found it 🤣

File: pkg/kubectl/cmd/util/openapi/validation/validation_test.go (hunk 1 above)
Comment (spzala) [reply pkg/kubectl/cmd/util/openapi/validation/validation_test.go]: @apelisse hi, thanks much for the quick review and suggestions :-). Here, I actually used one of the current tests as reference - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/openapi/validation/validation_test.go#L214  but let me see if I can use other tests. Thanks!

**Comment by apelisse** [comment]
/lgtm

Thank you for following up with that!

**Comment by apelisse** [comment]
@pwittrock can you approve that?

File: pkg/kubectl/cmd/util/openapi/validation/validation_test.go (hunk 2 above)
Comment (spzala) [reply pkg/kubectl/cmd/util/openapi/validation/validation_test.go]: LOL Thanks!!

**Comment by spzala** [comment]
@apelisse yrw and Thank you!! I am happily :) heading to a local kubernetes meet up in Raleigh now :-)

**Comment by pwittrock** [comment]
@spzala thanks!

**Comment by spzala** [comment]
Thanks @pwittrock !!

**Pull request merged**